txcache/wrappedTransaction.go

```diff
@@ -18,28 +18,27 @@ type WrappedTransaction struct {
 	ReceiverShardID uint32
 	Size            int64
 
-	PricePerUnit atomic.Uint64
+	// These fields are only set within "precomputeFields".
+	// We don't need to protect them with a mutex, since "precomputeFields" is called only once for each transaction.
+	// Additional note: "WrappedTransaction" objects are created by the Node, in dataRetriever/txpool/shardedTxPool.go.
+	Fee          *big.Int
+	PricePerUnit uint64
 }
 
 // precomputeFields computes (and caches) the (average) price per gas unit.
 func (wrappedTx *WrappedTransaction) precomputeFields(txGasHandler TxGasHandler) {
-	fee := txGasHandler.ComputeTxFee(wrappedTx.Tx).Uint64()
+	wrappedTx.Fee = txGasHandler.ComputeTxFee(wrappedTx.Tx)
 
 	gasLimit := wrappedTx.Tx.GetGasLimit()
-	if gasLimit == 0 {
-		return
+	if gasLimit != 0 {
+		wrappedTx.PricePerUnit = wrappedTx.Fee.Uint64() / gasLimit
 	}
-
-	wrappedTx.PricePerUnit.Store(fee / gasLimit)
 }
 
 // Equality is out of scope (not possible in our case).
 func (wrappedTx *WrappedTransaction) isTransactionMoreValuableForNetwork(otherTransaction *WrappedTransaction) bool {
-	// First, compare by price per unit
-	ppu := wrappedTx.PricePerUnit.Load()
-	ppuOther := otherTransaction.PricePerUnit.Load()
-	if ppu != ppuOther {
-		return ppu > ppuOther
+	if wrappedTx.PricePerUnit != otherTransaction.PricePerUnit {
+		return wrappedTx.PricePerUnit > otherTransaction.PricePerUnit
```

Comment:
before the txHash I would compare the gasLimits and return the one with the higher gasLimit.

Reply:
Good point! That somehow promotes less "execution fragmentation".

Reply:
yes!

Reply:
ok